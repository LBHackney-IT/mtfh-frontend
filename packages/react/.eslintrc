{
  "extends": "@hackney/eslint-config",
  "parserOptions": {
    "project": "./tsconfig.json"
  },
  "rules": {
    "prefer-arrow-callback": "off",
    "import/no-extraneous-dependencies": [
      "error",
      {
        "devDependencies": [
          "**/*/rollup.config.js",
          "**/*.test.ts",
          "**/*.test.tsx"
        ]
      }
    ],
    "jsx-a11y/anchor-is-valid": [
      "error",
      {
        "components": ["Link"],
        "aspects": ["invalidHref"]
      }
    ],
    "jsx-a11y/label-has-associated-control": "off"
    // "jsx-a11y/label-has-associated-control": [
    //   2,
    //   {
    //     "labelComponents": ["Label"],
    //     "labelAttributes": ["label"],
    //     "controlComponents": ["Input"],
    //     "assert": "htmlFor",
    //     "depth": 3
    //   }
    // ]
  }
}
