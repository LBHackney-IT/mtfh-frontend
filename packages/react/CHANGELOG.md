# @lbh-mtfh/react

## 1.1.0

### Minor Changes

- 37b7b3e: Initial release

### Patch Changes

- Updated dependencies [37b7b3e]
  - @lbh-mtfh/system@1.1.0
  - @lbh-mtfh/test-utils@1.1.0
