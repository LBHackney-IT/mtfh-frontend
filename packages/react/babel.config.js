const babel = require("../../babel.config")

module.exports = babel
