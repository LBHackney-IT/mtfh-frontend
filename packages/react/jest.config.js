const baseConfig = require("../../jest.config")

baseConfig.setupFilesAfterEnv.push("<rootDir>/jest.setup.ts")

module.exports = {
  ...baseConfig,
}
