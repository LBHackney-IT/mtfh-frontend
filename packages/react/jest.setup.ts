import * as LBHFrontend from "lbh-frontend"

window.LBHFrontend = LBHFrontend
