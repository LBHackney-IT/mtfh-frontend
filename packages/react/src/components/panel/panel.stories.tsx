import React from "react"

import { Panel } from "./panel"

export default {
  title: "Panel",
  component: Panel,
}

export const Base = () => <Panel titleText="Application complete" />
export const WithTitleAndBody = () => (
  <Panel
    titleText="Application complete"
    bodyText="Your reference number: HDJ2123F"
  />
)
export const WithTitle = () => <Panel titleText="Application complete" />
export const WithBody = () => (
  <Panel bodyText="Your reference number: HDJ2123F" />
)
