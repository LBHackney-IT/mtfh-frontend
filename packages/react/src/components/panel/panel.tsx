import React, { forwardRef } from "react"
import type * as Polymorphic from "@radix-ui/react-polymorphic"
import cn from "classnames"

import { widthOverrides } from "../../utils"

import { Heading } from "../heading"

import styles from "./styles.module.scss"

export interface PanelProps {
  titleText?: string
  bodyText?: string
  override?: number
  headingElementType?: "h1" | "h2" | "h3" | "h4" | "h5" | "h6"
}

export type PanelComponent = Polymorphic.ForwardRefComponent<"div", PanelProps>

export const Panel: PanelComponent = forwardRef(
  (
    {
      as: PanelComp = "div",
      titleText,
      bodyText,
      headingElementType = "h1",
      className,
      override,
      ...props
    },
    ref,
  ) => {
    const panelClasses = cn(
      styles.govukPanel,
      styles.govukPanelConfirmation,
      styles.lbhPanel,
      widthOverrides(override),
      className,
    )

    return (
      <PanelComp ref={ref} className={panelClasses} {...props}>
        {titleText && (
          <Heading as={headingElementType} className={styles.govukPanelTitle}>
            {titleText}
          </Heading>
        )}
        {bodyText && <div className={styles.govukPanelBody}>{bodyText}</div>}
      </PanelComp>
    )
  },
)
