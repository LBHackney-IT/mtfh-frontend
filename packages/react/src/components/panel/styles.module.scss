@import "~lbh-frontend/lbh/base";
@import "~lbh-frontend/lbh/components/lbh-panel/panel";

.govuk-panel--confirmation {

  &.panel__container--success, &.panel__container--error {
    display: flex;
    text-align: left;
    .panel-text__container {
      margin-left: 35px;
    }  
  }
  &.panel__container--error {
    background: lbh-colour("lbh-f02");
  }
  .icon-status__container {
    margin-top: 8px;
  }
  .panel-text__container {
    margin-top: 0;
  }

}