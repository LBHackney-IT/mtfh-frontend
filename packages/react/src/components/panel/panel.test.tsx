import React from "react"
import { testA11y } from "@hackney/mtfh-test-utils"
import { render, screen } from "@testing-library/react"

import { Panel } from "./panel"

test("it renders correctly", async () => {
  const { container } = render(<Panel />)
  expect(container).toMatchSnapshot()
  await testA11y(container)
})

test("it shows a title", () => {
  const TITLE_TEXT = "A title"
  render(<Panel titleText={TITLE_TEXT} />)
  expect(screen.getByText(TITLE_TEXT))
})

test("it shows a title that its polymorphic", () => {
  const TITLE_TEXT = "A title"
  const TITLE_TYPE = "h2"
  const { container } = render(
    <Panel titleText={TITLE_TEXT} headingElementType={TITLE_TYPE} />,
  )
  expect(container).toMatchSnapshot()
})

test("it shows a body", () => {
  const BODY_TEXT = "A title"
  render(<Panel bodyText={BODY_TEXT} />)
  expect(screen.getByText(BODY_TEXT))
})
