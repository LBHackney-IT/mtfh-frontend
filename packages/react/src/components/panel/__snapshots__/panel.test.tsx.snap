// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<div>
  <div
    class="govukPanel govukPanelConfirmation lbhPanel"
  />
</div>
`;

exports[`it shows a title that its polymorphic 1`] = `
<div>
  <div
    class="govukPanel govukPanelConfirmation lbhPanel"
  >
    <h2
      class="lbh-heading-h1 govukPanelTitle"
    >
      A title
    </h2>
  </div>
</div>
`;
